bcache/get.go

```diff
@@ -0,0 +1,310 @@
+// Copyright 2017 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package bcache
+
+import (
+	"bufio"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path"
+	"path/filepath"
+	"strconv"
+	"strings"
+)
+
+// parsePseudoFloat parses the peculiar format produced by bcache's bch_hprint
+func parsePseudoFloat(str string) (float64) {
+	ss := strings.Split(str, ".")
+
+	intPart, _ := strconv.ParseFloat(ss[0], 64)
```

Comment:
Annoying as it can be to propagate errors through your code, you'll want to check every string parsing error in this file.